### PR TITLE
Forbid invalid relabel configurations

### PR DESCRIPTION
**Pull request by hashmap**
This fix adds check if target_label value is set in case if action is replace or
hashmod
Issue [#1900]

### CODE REVIEW

File: config/testdata/target_label_missing.bad.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+scrape_configs:
```
Comment (brian-brazil) [inline config/testdata/target_label_missing.bad.yml]: You also need to test hashmod

Comment (hashmap) [inline config/testdata/target_label_missing.bad.yml]: Done

**Comment by brian-brazil** [comment]
Can you squash your commits?

**Comment by grobie** [comment]
Awesome, thanks!

**Pull request merged**